### ISSUE TITLE
Does not work when combined with @flaky

### ISSUE THREAD

**Issue by Chuckv** [astraw38/pytest-assume#14]
This plugin does not work for tests which are being re-run on assert failure via the Flaky plugin.  

Install flaky plugin and then try the following

```
from flaky import flaky

@flaky()
def test_error(self):
    pytest.assume(False)
```

The result that I got:

```
===Flaky Test Report===

test_error passed 1 out of the required 1 times. Success!

===End Flaky Test Report===

========================= no tests ran in 3.06 seconds =========================
```

**Comment by astraw38** [comment]
That's because it uses a hook that is earlier in the test process than pytest-assume. I believe that should be fixable -- I'll test something locally.

**Comment by astraw38** [comment]
As an alternative, pytest-rerunfailures definitely works with it. It'll take some time to sort out what flaky is doing though.

**Comment by astraw38** [comment]
After further review, flaky will *only* rerun if a test raises an exception. Most of the time this is indeed the case, but for pytest.assume we *don't* want to raise an exception during the test -- instead we complete the test & set the report to failed afterwards. 

I'll look into whether this is possible to be fixed... but no guarantees.

**Comment by Chuckv** [comment]
at my workplace we have several hundred tests that use @flakey decorators (mostly I think due to sauce and networking issues) so it would be really nice if this could work with that.. I don't see us wanting to change over to pytest-rerunfailures unless it also offers advantages over flakey

**Comment by astraw38** [comment]
Can you give the branch https://github.com/astraw38/pytest-assume/tree/use_new_hook here a trial to see if it works for your setup?

I did my best to verify locally w/ flaky, but it's got some bleedthrough with pytester (and doesn't fully seem to fully disable itself even with ``-p no:flaky`).

**Comment by Chuckv** [comment]
how do I make use of that branch?  (sorry, not very familiar with pytest plugins)  is there a way to install it with pip?  (I'm a bit of a newbie when it comes to python)

**Comment by astraw38** [comment]
``pip install git+https://github.com/astraw38/pytest-assume@use_new_hook``

It should install pytest-assume version 2.0.0 (which you can use to verify via ``pip show pytest-assume``)

**Comment by Chuckv** [comment]
Glad to have a look at it.   I'm presuming I can't pip install from that branch.  can you direct me to instructions for how I'd get the plugin installed from the branch so that I could test it?   I'm somewhat new to python, and never done anything of this sort before.  (working with pytest plugins in this way)

**Comment by Chuckv** [comment]
Oh thanks, so I can use pip, cool.  will try to have a look shortly..

**Comment by astraw38** [comment]
Closing -- merged in #16 which should fix this.
